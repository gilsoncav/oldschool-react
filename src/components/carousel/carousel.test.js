import Carousel from './carousel';

describe('Carousel component', () => {
  it('renders correctly', () => {
    const snapshot = snapRender(<Carousel />);
    expect(snapshot).toMatchSnapshot();
  });

  it('should have a default image when there is no prop', () => {
    const wrapper = render(<Carousel />);

    expect(
      wrapper.container.querySelector(
        `img[id="previous-image"][src="${process.env.PUBLIC_URL}/image/super-mario-kart.png"]`,
      ),
    ).toBeTruthy();

    expect(
      wrapper.container.querySelector(
        `img[id="current-image"][src="${process.env.PUBLIC_URL}/image/super-mario-world.jpg"]`,
      ),
    ).toBeTruthy();

    expect(
      wrapper.container.querySelector(
        `img[id="next-image"][src="${process.env.PUBLIC_URL}/image/top-gear.jpg"]`,
      ),
    ).toBeTruthy();
  });

  it('should call rollCarousel() every 1 second', () => {
    const clock = sinon.useFakeTimers();
    const wrapper = render(<Carousel intervalSeconds={1} />);
    expect(
      wrapper.container.querySelector(
        'img[id="previous-image"][data-index="0"]',
      ),
    ).toBeTruthy();

    act(() => {
      clock.tick(1000);
    });

    expect(
      wrapper.container.querySelector(
        'img[id="previous-image"][data-index="1"]',
      ),
    ).toBeTruthy();

    act(() => {
      clock.tick(1000);
    });

    expect(
      wrapper.container.querySelector(
        'img[id="previous-image"][data-index="2"]',
      ),
    ).toBeTruthy();
  });
});
