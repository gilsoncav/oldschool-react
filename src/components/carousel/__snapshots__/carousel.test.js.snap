// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component renders correctly 1`] = `
<div
  id="animated-carousel"
>
  <img
    alt="Random Game"
    data-index={0}
    id="previous-image"
    src="/image/super-mario-kart.png"
  />
  <img
    alt="Random Game"
    data-index={1}
    id="current-image"
    src="/image/super-mario-world.jpg"
  />
  <img
    alt="Random Game"
    data-index={2}
    id="next-image"
    src="/image/top-gear.jpg"
  />
</div>
`;
